providers/inwx/inwxProvider.go

```diff
@@ -33,18 +33,15 @@ Additional settings available in `creds.json`:
 
 */
 
-// inwxAPI is a thin wrapper around goinwx.Client.
-type inwxAPI struct {
-	client  *goinwx.Client
-	sandbox bool
-}
-
 // InwxDefaultNs contains the default INWX nameservers.
-var InwxDefaultNs = []string{"ns.inwx.de", "ns2.inwx.de", "ns3.inwx.eu"}
+var InwxProductionDefaultNs = []string{"ns.inwx.de", "ns2.inwx.de", "ns3.inwx.eu"}
 
 // InwxSandboxDefaultNs contains the default INWX nameservers in the sandbox / OTE.
 var InwxSandboxDefaultNs = []string{"ns.ote.inwx.de", "ns2.ote.inwx.de"}
 
+// InwxDomainIndex contains all domains saved in the INWX nameserver
+var InwxDomainIndex map[string]int
```

Comment:
I like the idea of caching the domains since dnscontrol will possibly call `EnsureDomainExists` a lot. 

What do you think about instead putting this into the `inwxAPI` struct? There might be some users who manage two separate INWX accounts with dnscontrol.

Reply:
A specific domain can only be registered/used in one INWX account, so even when multiple accounts are used, it should be good. But yes, it would be cleaner the way you suggested it.

Reply:
I've reworked it.